### PR TITLE
Added parser_ACTRIS-CCRES

### PR DESCRIPTION
**Pull request by KimCandolfi**
# Prework

- [x] I understand and agree to this repository's [code of conduct](https://github.com/ltelab/disdrodb/blob/main/CODE_OF_CONDUCT.md).
- [x] I understand and agree to this repository's [contributing guidelines](https://github.com/ltelab/disdrodb/blob/main/CONTRIBUTING.md).
- [x] I have already submitted an [issue](https://github.com/ltelab/disdrodb/issues) or [discussion thread](https://github.com/ltelab/disdrodb/discussions) to discuss my idea with the maintainers.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [x] Other, please describe: New parser

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and communicate accordingly:

**The PR fulfills these requirements:**

- [x] It's submitted to the branch named as follow : 
  - Add a reader: `reader-<institute>-<campaign>`
  - Fix a bug: `bugfix-<some_key>-<word>`
  - Improve the doc: `doc-<some_key>-<word>`
  - Add a new feature: `feature-<some_key>-<word>`
  - Refactor some code: `refactor-<some_key>-<word>`
  - Optimize some code: `optimize-<some_key>-<word>`
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] Don't forget to link PR to issue if you are solving one.
- [ ] All tests are passing.
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

# Related GitHub issues and pull requests

- Ref: #

# Summary

For now work only in lazy = False, I have hard code the flag in run_L0 function into parser_ACTRIS-CCRES.py.
In the future I will fix also with lazy = True

### CODE REVIEW

**Comment by KimCandolfi** [comment]
Raw data sample file: [spectropluvio_0a_Lz1LpnF60secPrainratePsize_v01_20220804_000000_1440.txt](https://github.com/ltelab/disdrodb/files/9555310/spectropluvio_0a_Lz1LpnF60secPrainratePsize_v01_20220804_000000_1440.txt)

**Comment by regislon** [comment]
@KimCandolfi could you please asign somebody to review the PR ?

**Comment by KimCandolfi** [comment]
Do you can rewiew my PR please? Thank you @ghiggi

**Comment by regislon** [comment]
Hi @KimCandolfi, on the top right corner of this Pull Request, you can chose the reviewer. He will be automatically informed.

**Comment by ghiggi** [comment]
I will review it on Friday ;)

File: disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+    def df_sanitizer_fun(df, lazy=False):
+        # Import dask or pandas
+        if lazy:
+            import dask.dataframe as dd
+        else:
+            import pandas as dd
+
+        # Remove header columns
+        df = df[
```
Comment (ghiggi) [inline disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py:217]: The header appear just at the beginning of the file right? 
So I think the header column can be removed through some `reader_kwargs ` light `skiprows `. Please correct for this

File: disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+        # Parse time
+        df["time"] = df["date"] + "-" + df["time_temp"]
+        df["time"] = dd.to_datetime(df["time"], format="%Y/%m/%d-%H:%M:%S")
+        df = df.drop(columns=["date", "time_temp"])
+
+        # Set NaN into <SPECTRUM>ZERO</SPECTRUM> in raw_drop_number
+        import numpy as np
+
+        df["raw_drop_number"] = df["raw_drop_number"].replace(
```
Comment (ghiggi) [inline disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py:261]: I guess this occurs when no drop detected right? Document with a comment

File: disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+            "<SPECTRUM>ZERO</SPECTRUM>", np.NaN
+        )
+
+        # Cannot implement dask for this loop, so only pandas for now
+        for i, v in df.iterrows():
+            # Check if v is NaN
+            if not v["raw_drop_number"] != v["raw_drop_number"]:
+
+                v["raw_drop_number"] = v["raw_drop_number"].replace("<SPECTRUM>", "")
```
Comment (ghiggi) [inline disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py:270]: These two rows can be placed outside the loop !

File: disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+                v["raw_drop_number"] = v["raw_drop_number"].replace("</SPECTRUM>", "")
+
+                # Add 0 digits to raw_drop_number
```
Comment (ghiggi) [inline disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py:273]: The code here below could be delegated to the inner processing code. 
I will work on that on Friday ;)

File: disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+        )
+
+        return df
+
+    ##------------------------------------------------------------------------.
+    #### - Define glob pattern to search data files in raw_dir/data/<station_id>
+    files_glob_pattern = "*.txt*"  # There is only one file without extension
```
Comment (ghiggi) [inline disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py:292]:
```suggestion
    files_glob_pattern = "*.txt"  # There is only one file without extension
```

**Review by ghiggi — COMMENTED** [review]
@KimCandolfi please address my comments ;)

File: disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+        ]
+
+        # Split into columns and assign name
+        df = df["TO_SPLIT"].str.split(";", expand=True, n=16)
+
+        column_names = [
+            "date",
+            "time_temp",
+            "rainfall_rate_32bit",
+            "precipitation_since_start_TO_DROP",
```
[the remark this answers is not in the document]
Comment (KimCandolfi) [reply disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py:230]: Yes, make sense, I will change it

File: disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+            "reflectivity_32bit",
+            "mor_visibility",
+            "laser_amplitude",
+            "number_particles",
+            "sensor_temperature",
+            "sensor_heating_current",
+            "sensor_battery_voltage",
+            "rain_kinetic_energy",
```
[the remark this answers is not in the document]
Comment (KimCandolfi) [reply disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py:238]: Yes alright

File: disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py (hunk 4 above)
Comment (KimCandolfi) [reply disdrodb/L0/readers/FRANCE/parser_ACTRIS-CCRES.py:273]: ok

**Review by ghiggi — APPROVED** [review]
LGTM now. Ready to be merged. 
I will try to further simplify the sanitizer on Friday by making modifications to the `L0_processing.py` functions.

**Pull request merged**